ext/debug/ctf/bottoms/posix/ctf_bottom.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018 Oticon A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _CTF_BOTTOM_POSIX_H
+#define _CTF_BOTTOM_POSIX_H
+
+#include <stddef.h>
+#include <string.h>
+#include <stdio.h>
+#include <zephyr/types.h>
+#include <ctf_map.h>
+
+
+/* Obtain a field's size at compile-time.
+ * Internal to this bottom-layer.
+ */
+#define CTF_BOTTOM_INTERNAL_FIELD_SIZE(x)      + sizeof(x)
+
+/* Append a field to current event-packet.
+ * Internal to this bottom-layer.
+ */
+#define CTF_BOTTOM_INTERNAL_FIELD_APPEND(x)			\
+	{							\
+		memcpy(epacket_cursor, &(x), sizeof(x));	\
```

Comment:
IMHO using memcpy is an overkill as most of arguments are 4-bytes or less. We can use  __builtin_memcpy() and/or other tricks (like `switch (sizeof(x)) { ... }`) to create faster code.